src/concrete/ml/onnx/ops_impl.py

```diff
@@ -893,11 +897,19 @@ def numpy_equal(
     Args:
         x (numpy.ndarray): Input tensor
         y (numpy.ndarray): Input tensor
+        lsbs_to_remove (Optional[int]): The number of the least significant bits to remove
 
     Returns:
         Tuple[numpy.ndarray]: Output tensor
     """
 
+    if lsbs_to_remove is not None and lsbs_to_remove > 0:
+        # opt 1 :R == D <=> R >= D
+        # opt 2: R == D <=> R > D - 1 (better)
+        return rounded_comparison(
+            y, x, lsbs_to_remove, operation=lambda x: x >= 0
```

Comment:
No, all the comparison operators have been adapted to become equivalent for rounding.
Because rounding doesn't work as a truncate.
x == y <=> y >= x, because y is the 'max' sum.